logutils/namespaces_tree_test.go

```diff
@@ -0,0 +1,102 @@
+package logutils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+)
+
+func TestBuildNamespacesTree(t *testing.T) {
+	tests := []struct {
+		name       string
+		namespaces string
+		minLevel   zapcore.Level
+		err        error
+	}{
+		{
+			name:       "valid namespaces",
+			namespaces: "namespace1:debug,namespace2.namespace3:error",
+			minLevel:   zapcore.DebugLevel,
+			err:        nil,
+		},
+		{
+			name:       "invalid format",
+			namespaces: "namespace1:debug,namespace2.namespace3",
+			minLevel:   zapcore.DebugLevel,
+			err:        errInvalidNamespacesFormat,
+		},
+		{
+			name:       "empty namespaces",
+			namespaces: "",
+			minLevel:   zapcore.InvalidLevel,
+			err:        nil,
+		},
+		{
+			name:       "invalid level",
+			namespaces: "namespace1:invalid",
+			minLevel:   zapcore.InvalidLevel,
+			err:        errInvalidNamespacesFormat,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tree, err := buildNamespacesTree(tt.namespaces)
+			require.ErrorIs(t, err, tt.err)
+			if tt.err == nil {
+				require.NotNil(t, tree)
+			}
+		})
+	}
+}
+
+func TestLevelFor(t *testing.T) {
+	tree, err := buildNamespacesTree("namespace1:error,namespace1.namespace2:debug,namespace1.namespace2.namespace3:info")
```

Comment:
In this case for `namespace2` there will be no match, right?

Reply:
Right :+1:

Reply:
https://github.com/status-im/status-go/blob/2bb5cae32985e52375c8f4dd715e7a83a4a9e3b5/logutils/namespaces_tree_test.go#L89-L93